### PR TITLE
add updated attribute to list all versions, add ability to list individu...

### PR DESCRIPTION
**Pull request by pdmars**
...al versions

### CODE REVIEW

File: reddwarf/versions.py [hunk 1]
```diff
@@ -22,35 +22,64 @@
 from reddwarf.common import wsgi
 
 
+VERSIONS = {
+    "1.0": {
+        "id": "v1.0",
+        "status": "CURRENT",
+        "updated": "2012-01-01T00:00:00Z",
+        "links": [
+        ],
```
Comment (jcru) [inline reddwarf/versions.py]: you could close the above list brackets and save a line (think about the electronic trees :)

File: reddwarf/versions.py [hunk 2]
```diff
@@ -22,35 +22,64 @@
...
+}
+
+
 class VersionsController(wsgi.Controller):
 
     def index(self, request):
-        """Respond to a request for all OpenStack API versions."""
-        versions = [Version("v1.0", "CURRENT", request.application_url)]
+        """Respond to a request for API versions."""
+        global VERSIONS
```
Comment (jcru) [inline reddwarf/versions.py]: global isn't necessary unless your modifying VERSIONS, which I do not believe you are.

File: reddwarf/versions.py [hunk 3]
```diff
@@ -22,35 +22,64 @@
...
+        versions = []
+        for key, data in VERSIONS.items():
+            v = BaseVersion(data["id"], data["status"],
+                        request.application_url, data["updated"])
+            versions.append(v)
         return wsgi.Result(VersionsDataView(versions))
 
+    def show(self, request):
+        """Respond to a request for a specific API version."""
+        global VERSIONS
```
Comment (jcru) [inline reddwarf/versions.py]: ditto as above.

**Comment by ed-** [comment]
If the only time we're going to deal with the data in VERSIONS is to create a BaseVersion, I think we should just build the [Base]Versions once and return those.

Take a look at reddwarf/instance/tasks.py to see what we do with our InstanceTask objects.

File: reddwarf/versions.py (hunk 2 above)
Comment (pdmars) [inline reddwarf/versions.py]: Yeah, true. This is one of those things about Python that I'm not a fan of.

File: reddwarf/versions.py (hunk 1 above)
Comment (pdmars) [inline reddwarf/versions.py]: I thought the value of my contributions was judged by LOC.

File: reddwarf/versions.py [hunk 4]
```diff
@@ -22,35 +22,64 @@
...
         self.base_url = base_url
+        self.updated = updated
 
     def data(self):
         return dict(id=self.id,
             status=self.status,
-            links=[dict(rel="self",
-                href=self.url())])
+            updated=self.updated,
+            links=[dict(rel="self", href=self.url())])
```
Comment (ed-) [inline reddwarf/versions.py:66]:
``` python
dict()
```

what's wrong with `{}`? Works just fine everywhere else in your changes.

Comment (pdmars) [inline reddwarf/versions.py:66]: Nothing's wrong with it, that's just what was there before, I just removed the line break that I didn't like looking at.

Comment (ed-) [inline reddwarf/versions.py:66]: I'll agree the linebreak was goofy

**Comment by jcru** [comment]
+1

**Comment by cp16net** [comment]
+1

**Comment by ed-** [comment]
+1

File: reddwarf/versions.py [hunk 5]
```diff
@@ -22,35 +22,61 @@
 from reddwarf.common import wsgi
 
 
+VERSIONS = {
+    "1.0": {
+        "id": "v1.0",
+        "status": "CURRENT",
+        "updated": "2012-01-01T00:00:00Z",
```
Comment (rnirmal) [inline reddwarf/versions.py]: lets set this to Aug 1.. 2012-08-01

File: reddwarf/versions.py (hunk 1 above)
Comment (rnirmal) [inline reddwarf/versions.py]: :)

File: reddwarf/versions.py (hunk 5 above)
Comment (pdmars) [inline reddwarf/versions.py]: Done.

**Comment by rnirmal** [comment]
+1

**Pull request merged**